modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/exp/ExpressionFactoryImplTest.java

```diff
@@ -704,8 +701,36 @@ public void testRowSource() {
         RexNode val10 = rexBuilder.makeExactLiteral(new BigDecimal("1"), intType);
         RexNode val11 = rexBuilder.makeExactLiteral(new BigDecimal("2"), bigIntType);
```

Comment:
1. I think we need to ensure that DECIMAL values preserve precision/scale after this transformation.
2. Ideally, we need to check literals of all types here.

Reply:
Thanks, the test has been improved, please take a look again.